config/config.go

```diff
@@ -119,6 +162,48 @@ func (c *MenderShellConfig) Validate() error {
 		}
 	}
 
+	//check if shell is given, if not, defaulting to /bin/sh
+	if c.ShellCommand == "" {
+		log.Warnf("ShellCommand is empty, defaulting to %s", DefaultShellCommand)
+		c.ShellCommand = DefaultShellCommand
+	}
+
+	if !filepath.IsAbs(c.ShellCommand) {
+		return errors.New("given shell (" + c.ShellCommand + ") is not an absolute path")
+	}
+
+	if !isExecutable(c.ShellCommand) {
+		return errors.New("given shell (" + c.ShellCommand + ") is not executable")
+	}
+
+	if c.User == "" {
+		return errors.New("please provide a user to run the shell as")
+	}
+	u, err := user.Lookup(c.User)
+	if err == nil && u == nil {
+		return errors.New("unknown error while getting a user id")
+	}
+	if err != nil {
+		return err
+	}
+
+	if !isInShells(c.ShellCommand) {
+		log.Warnf("ShellCommand %s is not present in /etc/shells", DefaultShellCommand)
```

Comment:
Here should be `c.ShellCommand`
```suggestion
		log.Warnf("ShellCommand %s is not present in /etc/shells", c.ShellCommand)
```

Reply:
@tranchitella ok, so what do you have in mind? do you want to enforce the shell to be present in /etc/shells and return with error here if not? right now I chose a "soft" solution of printing a warning.

Reply:
I chose to exit with an error to address this comment.